### PR TITLE
chore: Changes only latest version in SDK update

### PR DESCRIPTION
**Pull request by lantoli**
## Proposed changes

Changes only latest version in SDK update

_Jira ticket:_ CLOUDP-276362

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: .github/workflows/update-sdk.yml [hunk 1]
```diff
@@ -14,14 +14,15 @@ jobs:
         with:
           go-version-file: 'cfn-resources/go.mod'
       - name: Update files
-        run:  make tools update-atlas-sdk
+        run:  make tools update-atlas-sdk generate-mocks
```
Comment (lantoli) [inline .github/workflows/update-sdk.yml:17]: generate mocks in case mock versions changed

File: .github/workflows/update-sdk.yml [hunk 2]
```diff
@@ -14,14 +14,15 @@ jobs:
...
       - name: Verify Changed files
         uses: tj-actions/verify-changed-files@bbd436f8e34e14fee2a0f7bc5b14bcc6b40927cf
         id: verify-changed-files
       - name: Create PR
         uses: peter-evans/create-pull-request@5e914681df9dc83aa4e4905692ca88beb2f9e91f
         if: steps.verify-changed-files.outputs.files_changed == 'true'
         with:
+          token: ${{ secrets.APIX_BOT_PAT }}
```
Comment (lantoli) [inline .github/workflows/update-sdk.yml:25]: to have PR checks run

File: cfn-resources/scripts/update-sdk.sh [hunk 3]
```diff
@@ -16,15 +16,24 @@
...
 LATEST_SDK_RELEASE=$(echo "${LATEST_SDK_TAG}" | cut -d '.' -f 1)
-echo "==> Updating SDK to latest major version ${LATEST_SDK_TAG}"
-gomajor get "go.mongodb.org/atlas-sdk/${LATEST_SDK_RELEASE}@${LATEST_SDK_TAG}"
+echo "LATEST_SDK_RELEASE: $LATEST_SDK_RELEASE"
+echo "==> Updating SDK ${CURRENT_SDK_RELEASE} to latest major version ${LATEST_SDK_TAG}"
+
+gomajor get --rewrite "go.mongodb.org/atlas-sdk/${CURRENT_SDK_RELEASE}" "go.mongodb.org/atlas-sdk/${LATEST_SDK_RELEASE}@${LATEST_SDK_TAG}"
 go mod tidy
 
-LATEST_SDK_STRIPPED_MAYOR_VERSION="${LATEST_SDK_RELEASE%%.*}"
```
Comment (lantoli) [inline cfn-resources/scripts/update-sdk.sh:26]: LATEST_SDK_RELEASE is already in required format

**Review by EspenAlbert — APPROVED** [review]
Nice

**Pull request merged**